File: Professional/EndUserDesigner/VB.NET/EndUserDesigner/ExportHelper.vb
Imports GrapeCity.ActiveReports.Extensibility.Rendering
Imports GrapeCity.ActiveReports.Document
Imports GrapeCity.ActiveReports.Export
Public Class SectionExport
	Public Property DisplayName() As String
		Get
			Return _displayName
		End Get
		Set(value As String)
			_displayName = value
		End Set
	End Property
	Private _displayName As String
	Public Property Filter() As String
		Get
			Return _filter
		End Get
		Set(value As String)
			_filter = value
		End Set
	End Property
	Private _filter As String
	Public Property DefaultFileName() As String
		Get
			Return _defaultFileName
		End Get
		Set(value As String)
			_defaultFileName = value
		End Set
	End Property
	Private _defaultFileName As String
	Public Property Settings() As IDocumentExportEx
		Get
			Return _settings
		End Get
		Set(value As IDocumentExportEx)
			_settings = value
		End Set
	End Property
	Private _settings As IDocumentExportEx
	Public Sub Export(sectionDocument As SectionDocument, filePath As String)
		Settings.Export(sectionDocument, filePath)
	End Sub
End Class
Public Class PageExport
	Public Property DisplayName() As String
		Get
			Return _displayName
		End Get
		Set(ByVal value As String)
			_displayName = value
		End Set
	End Property
	Private _displayName As String
	Public Property Filter() As String
		Get
			Return _filter
		End Get
		Set(ByVal value As String)
			_filter = value
		End Set
	End Property
	Private _filter As String
	Public Property RenderingExtension() As IRenderingExtension
		Get
			Return _renderingExtension
		End Get
		Set(ByVal value As IRenderingExtension)
			_renderingExtension = value
		End Set
	End Property
	Private _renderingExtension As IRenderingExtension
	Public Property DefaultFileName() As String
		Get
			Return _defaultFileName
		End Get
		Set(ByVal value As String)
			_defaultFileName = value
		End Set
	End Property
	Private _defaultFileName As String
	Public Property Settings() As ISettings
		Get
			Return _settings
		End Get
		Set(ByVal value As ISettings)
			_settings = value
		End Set
	End Property
	Private _settings As ISettings
End Class

File: Professional/EndUserDesigner/C#/EndUserDesigner/ExportHelper.cs
using GrapeCity.ActiveReports.Extensibility.Rendering;
using GrapeCity.ActiveReports.Document;
using GrapeCity.ActiveReports.Export;
namespace GrapeCity.ActiveReports.Samples.EndUserDesigner
{
	class SectionExport
	{
		public string DisplayName { get; set; }
		public string Filter { get; set; }
		public string DefaultFileName { get; set; }
		public IDocumentExportEx Settings { get; set; }
		public void Export(SectionDocument sectionDocument, string filePath)
		{
			Settings.Export(sectionDocument, filePath);
		}
	}
	class PageExport
	{
		public string DisplayName { get; set; }
		public string Filter { get; set; }
		public IRenderingExtension RenderingExtension { get; set; }
		public string DefaultFileName { get; set; }
		public ISettings Settings { get; set; }
	}
}
